android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java

```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
+            // This is a legal point to keep posting edgeEvents updates, as the EdgeEventBus
+            // should now be initalized, unless disabled.
+
+            registerClientReplyFuture =
+                    me.registerClientFuture(registerClientRequest,
+                            dmeHostAddress, port, 10000);
+            registerClientReply = registerClientReplyFuture.get();
+            Log.i(TAG, "Register status: " + registerClientReply.getStatus());
+            AppClient.VerifyLocationRequest verifyRequest =
+                    me.createDefaultVerifyLocationRequest(ctx, location)
+                            .build();
+            Log.i(TAG, "verifyRequest is " + verifyRequest);
+
+            if (verifyRequest != null) {
+                // Location Verification (Blocking, or use verifyLocationFuture):
+                AppClient.VerifyLocationReply verifiedLocation =
+                        me.verifyLocation(verifyRequest, dmeHostAddress, port, 10000);
+                Log.i(TAG, "VerifyLocationReply is " + verifiedLocation);
+
+                someText += "[Location Verified: Tower: " + verifiedLocation.getTowerStatus() +
+                        ", GPS LocationStatus: " + verifiedLocation.getGpsLocationStatus() +
+                        ", Location Accuracy: " + verifiedLocation.getGpsLocationAccuracyKm() + " ]\n";
+                List<distributed_match_engine.Appcommon.AppPort> ports = closestCloudlet.getPortsList();
+                String portListStr = "";
+                boolean first = true;
+                String appPortFormat = "{Protocol: %d, Container Port: %d, External Port: %d, Path Prefix: '%s'}";
+                for (Appcommon.AppPort aPort : ports) {
+                    if (!first) {
+                        portListStr += ", ";
 
-                            }
-                            appInstListText += "]]";
-                        }
-                        if (!appInstListText.isEmpty()) {
-                            someText += appInstListText;
-                        }
-                    } else {
-                        someText = "Cannot create request object.\n";
-                        if (!locationVerificationAllowed) {
-                            someText += " Reason: Enhanced location is disabled.\n";
+                    }
+                    portListStr += String.format(Locale.getDefault(), appPortFormat,
+                            aPort.getProto().getNumber(),
+                            aPort.getInternalPort(),
+                            aPort.getPublicPort(),
+                            aPort.getEndPort());
+
+                    //String l7Url = mMatchingEngine.getAppConnectionManager().createUrl(closestCloudlet, aPort, aPort.getPublicPort(), "http", null);
+
+                    String host = aPort.getFqdnPrefix() + closestCloudlet.getFqdn();
+                    int knownPort = 8008;
```

Comment:
I think 8008 is CV, You can use any of the ports here https://mobiledgex.atlassian.net/wiki/spaces/SWDEV/pages/1615101953/SDK+Test+App

Reply:
Garner is the port here is correct?